src/utilities/viewlift/requests/viewliftListUserWatched.ts

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable consistent-return */
+import axios from 'axios';
+import { IListUserWatched, IListUserWatchedReturn, IWatchedStream } from '../../../interfaces';
+import { listUserWatchedQuery } from '../gqlQueries';
+import CustomError from '../../CustomError';
+
+const getAllWatches = async (
+  listUserWatchedData: {
+    Authorization: string,
+    viewliftEndpoint: string,
+    counter: number,
+    offset: number,
+    limit: number,
+    watchTimeFrom: string,
+    watchTimeTo: string,
+  },
+): Promise<IListUserWatchedReturn> => {
+  let watchesList: IWatchedStream[] = [];
+  let allWatchesCount = 0;
+  const listUserWatched = async (
+    {
+      Authorization,
+      viewliftEndpoint,
+      offset = 0,
+      limit,
+      watchTimeFrom,
+      watchTimeTo,
+    }: IListUserWatched,
+  ): Promise<IListUserWatchedReturn> => {
+    try {
+      const { data: { data: { listUserWatched: { totalCount, watchedStreams } } } } = await axios
+        .post(viewliftEndpoint, {
+          query: listUserWatchedQuery,
+          variables: {
+            watchTimeFrom,
+            watchTimeTo,
+            limit,
+            offset,
+          },
+        }, {
+          headers: {
+            Authorization,
+          },
+        });
+
+      let { counter } = listUserWatchedData;
+
+      allWatchesCount = totalCount;
+      watchesList = [...watchesList, ...watchedStreams];
+
+      if (totalCount > offset + limit) {
+        await listUserWatched({
+          watchTimeFrom,
+          watchTimeTo,
+          limit,
+          offset: offset + limit,
+          Authorization,
+          viewliftEndpoint,
+        });
+      }
+
+      const isTotalMatch = totalCount === watchesList.length;
+      if (!isTotalMatch) {
+        if (counter < 10) {
+          counter += 1;
+          return await getAllWatches({
+            counter,
+            Authorization,
+            viewliftEndpoint,
+            limit,
+            offset: 0,
+            watchTimeFrom,
+            watchTimeTo,
+          });
+        }
+        throw new Error('trails limit exceeded');
+      }
+
+      return {
+        totalCount,
+        watchedStreams: watchesList,
+      };
+    } catch (err: any) {
+      throw new CustomError(err.message, 'Error while fetching userWatched data');
+    }
+  };
+
```

Comment:
Same as above. Try not to use any type and fix eslint errors
```suggestion
   
    } catch (err) {
      if (err instanceof Error) {
        throw new CustomError(err.message, 'Error while fetching userWatched data');
      }
      throw err;
    }

```